src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -32,6 +32,8 @@ public function __construct(array $decoders = array())
 
     /**
      * {@inheritdoc}
+     * 
+     * @throws RuntimeException If no decoder is found.
```

Comment:
>Also, I don't understand what's the point of specifying runtime exceptions in the @throws tag. By analogy with unchecked exceptions from Java, these exceptions are mostly for an errors: no decoder found, bad configuration, bug, etc. User of an interface doesn't catch them, because he can't "fix" it. He has to fix the code, configuration, etc. in order to resolve his problem.

Overall I agree, but here for example you have a `getDecoder()` throwing a `RuntimeException`, forcing you to catch it to be used in `supports()`. Maybe it should not be a `RuntimeException` in the first place, but anyway I think it's worth documenting as long as it's a specific exception like here (it's not a `\RuntimeException` – well it is by inheritance, but  a serializer one).